src/tools/cephfs_mirror/ServiceDaemon.cc

```diff
@@ -68,11 +68,10 @@ int ServiceDaemon::init() {
     id = id.substr(CEPHFS_MIRROR_AUTH_ID_PREFIX.size());
   }
   std::string instance_id = stringify(m_rados->get_instance_id());
-
   std::map<std::string, std::string> service_metadata = {{"id", id},
                                                          {"instance_id", instance_id}};
-  int r = m_rados->service_daemon_register("cephfs-mirror", instance_id,
-                                           service_metadata);
+  string name = cct->_conf->name.get_id();
+  int r = m_rados->service_daemon_register("cephfs-mirror", name, service_metadata);
```

Comment:
Why don't we keep `instance_id` just like rbd-mirror does?

Reply:
Like `rbd-mirror.$INSTANCE_ID` IIRC.

Reply:
if rbd-mirror is doing this then it is also broken :(, will take a look

Reply:
Actually rbd-mirror works because it is named predictably and cephadm behaves accordingly.  I can do the same thing for cephfs-mirror.